### PR TITLE
fixed bug in resolver when rejoining subnet

### PR DESCRIPTION
**Pull request by adlrocha**
Fix bug for which pubsub topic validator of resolver was not being unregistered when leaving or closing a subnet. This PR fixes the issue.

### CODE REVIEW

**Comment by adlrocha** [comment]
@dnkolegov, this is a minor fix, but pinging you just for sanity-check.

File: chain/consensus/hierarchical/subnet/manager/manager.go [hunk 1]
```diff
@@ -318,7 +318,8 @@ func (s *SubnetMgr) Close(ctx context.Context) error {
 			return err
 		}
 	}
-	return nil
+	// Close resolver
+	return s.r.Close()
```
Comment (dnkolegov) [inline chain/consensus/hierarchical/subnet/manager/manager.go:322]: Why do we return if an error happened? don't we need to continue?

File: chain/consensus/hierarchical/subnet/manager/subnet.go [hunk 2]
```diff
@@ -167,6 +167,8 @@ func (sh *Subnet) Close(ctx context.Context) error {
 	sh.syncer.Stop()
 	// Close message pool
 	err5 := sh.mpool.Close()
+	// Close resolver.
+	err6 := sh.r.Close()
```
Comment (dnkolegov) [inline chain/consensus/hierarchical/subnet/manager/subnet.go:171]: I believe err1, err2, ... are not idiomatic names for errors. Let's add TODO?

**Review by dnkolegov — APPROVED** [review]

File: chain/consensus/hierarchical/subnet/manager/subnet.go (hunk 2 above)
Comment (adlrocha) [reply chain/consensus/hierarchical/subnet/manager/subnet.go:171]: Good point, adding TODO.

**Pull request merged**